### PR TITLE
Qt6: fix build on mingw-w64

### PR DESCRIPTION
**Pull request by oltolm**
Fixed the build on mingw-w64 and some compiler warnings.

### CODE REVIEW

File: rpcs3/Emu/Cell/lv2/sys_net.cpp [hunk 1]
```diff
@@ -257,8 +257,6 @@ lv2_socket::lv2_socket(utils::serial& ar, lv2_socket_type _type)
 	ar.pos += 8;
 #endif
 
-	const s32 version = GET_SERIALIZATION_VERSION(lv2_net);
```
Comment (elad335) [inline rpcs3/Emu/Cell/lv2/sys_net.cpp:260]: Do not remove, mark with [[maybe_unused]] instead.

Comment (oltolm) [reply rpcs3/Emu/Cell/lv2/sys_net.cpp:260]: Done.

**Comment by illusion0001** [comment]
Might be a good idea to setup circus to build on such compilers.

**Comment by oltolm** [comment]
ping

File: rpcs3/CMakeLists.txt [hunk 2]
```diff
@@ -89,8 +90,7 @@ target_link_libraries(rpcs3 PRIVATE ${ADDITIONAL_LIBS})
 
 # Unix display manager
 if(X11_FOUND)
-    target_include_directories(rpcs3 PUBLIC ${X11_INCLUDE_DIR})
-    target_link_libraries(rpcs3 PRIVATE ${X11_LIBRARIES})
+    target_link_libraries(rpcs3 PRIVATE X11::X11)
```
Comment (Megamouse) [inline rpcs3/CMakeLists.txt:93]: What does this have to do with mingw?

File: rpcs3/util/vm_native.cpp [hunk 3]
```diff
@@ -86,14 +86,6 @@ namespace utils
 	constexpr int c_mfd_huge_2mb = 0;
 #endif
 
-#ifndef MEM_RESERVE_PLACEHOLDER
```
Comment (Megamouse) [inline rpcs3/util/vm_native.cpp:89]: Why are these removed?

File: CMakeLists.txt [hunk 4]
```diff
@@ -105,18 +105,11 @@ if(CCACHE_FOUND)
     set(CMAKE_CXX_COMPILER_LAUNCHER ccache)
 endif()
 
-if(WIN32)
-    add_compile_definitions(UNICODE)
-    add_compile_definitions(_WIN32_WINNT=0x0602)
```
Comment (Megamouse) [inline CMakeLists.txt:110]: Why are these moved?

File: rpcs3/util/vm_native.cpp (hunk 3 above)
Comment (oltolm) [reply rpcs3/util/vm_native.cpp:89]: They have been added to mingw-w64.

File: rpcs3/CMakeLists.txt (hunk 2 above)
Comment (oltolm) [reply rpcs3/CMakeLists.txt:93]: Nothing. Do you want a separate PR?

File: CMakeLists.txt (hunk 4 above)
Comment (oltolm) [reply CMakeLists.txt:110]: Now these definitions are added to targets that require them and not globally.

File: rpcs3/CMakeLists.txt (hunk 2 above)
Comment (Megamouse) [reply rpcs3/CMakeLists.txt:93]: Since this is unrelated, it would make more sense for regression testing

Comment (oltolm) [reply rpcs3/CMakeLists.txt]: Reverted.
